src/evg_config.rs

```diff
@@ -0,0 +1,848 @@
+    ///
+    /// # Arguments
+    ///
+    /// * `task_def` - Task definition to query.
+    /// * `run_var` - Variable to query.
+    ///
+    /// # Returns
+    ///
+    /// Value of run_var for the given task, an `Error` will be returned if not defined.
+    fn lookup_required_param_bool(&self, task_def: &EvgTask, run_var: &str) -> Result<bool> {
+        Ok(match self.get_gen_task_var(task_def, run_var) {
+            Some(v) => v.parse()?,
+            _ => bail!(format!(
+                "Missing var '{}' for task '{}'",
+                run_var, task_def.name
+            )),
+        })
+    }
+
+    /// Lookup the given variable in the task definition.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_def` - Task definition to query.
+    /// * `run_var` - Variable to query.
+    /// * `default` - Default value to use if `run_var` is undefined.
+    ///
+    /// # Returns
+    ///
+    /// Value of run_var for the given task, the default will be returned if not defined.
+    fn lookup_default_param_bool(
+        &self,
+        task_def: &EvgTask,
+        run_var: &str,
+        default: bool,
+    ) -> Result<bool> {
+        Ok(match self.get_gen_task_var(task_def, run_var) {
+            Some(v) => v.parse()?,
+            _ => default,
+        })
+    }
+
+    /// Lookup the given variable in the task definition.
+    ///
+    /// # Arguments
+    ///
+    /// * `task_def` - Task definition to query.
+    /// * `run_var` - Variable to query.
+    /// * `default` - Default value to use if `run_var` is undefined.
+    ///
+    /// # Returns
+    ///
+    /// Value of run_var for the given task, the default will be returned if not defined.
+    fn lookup_default_param_str(&self, task_def: &EvgTask, run_var: &str, default: &str) -> String {
+        self.get_gen_task_var(task_def, run_var)
+            .unwrap_or(default)
+            .to_string()
+    }
+}
+
+/// Get the shrub function make the 'generate resmoke task' call in the given task.
+///
+/// # Arguments
+///
+/// * `task` - Shrub task to query.
+///
+/// # Returns
+///
+/// Function call to 'generate resmoke task'.
+fn get_generate_resmoke_func(task: &EvgTask) -> Option<&FunctionCall> {
+    let command = task.commands.iter().find(|c| {
+        if let Function(func) = c {
+            if func.func == GENERATE_RESMOKE_TASKS {
+                return true;
+            }
+        }
+        false
+    });
+
+    if let Some(Function(func)) = command {
+        Some(func)
+    } else {
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use maplit::btreemap;
+    use maplit::hashmap;
+    use shrub_rs::models::commands::{fn_call, fn_call_with_params};
+
+    use super::*;
+
+    // is_task_generated tests.
+
+    #[test]
+    fn test_is_task_generated_should_return_false_if_not_generated() {
+        let evg_task = EvgTask {
+            commands: vec![fn_call("hello world"), fn_call("run tests")],
+            ..Default::default()
```

Comment:
rust doesn't have default parameters, but uses this `Default::default()` instead. This is a trait you can define for types that provides some default values.